src/TableOfContents/index.tsx

```diff
@@ -36,24 +36,22 @@ export type ITableOfContentsLink = TableOfContentsItem & {
   isExternalLink?: boolean;
 };
 
-export type RowRendererType<T extends TableOfContentsItem> = (props: {
+export type RowComponentProps<T extends TableOfContentsItem> = {
   item: T;
-  key: number | string;
-  getProps: (
-    node: ITableOfContentsNode,
-  ) => {
-    onClick: ((e: React.MouseEvent<any, MouseEvent>) => void) | undefined;
-    style: React.CSSProperties;
-    className: string;
-  };
-  DefaultRow: React.FC<ITableOfContentsNode<T>>;
-}) => React.ReactElement | undefined;
+  index: number;
+  isExpanded: boolean;
+  toggleExpanded: () => void;
+};
+
+export type RowComponentType<T extends TableOfContentsItem> = React.ComponentType<RowComponentProps<T>>;
 
 export interface ITableOfContents<T extends TableOfContentsItem = TableOfContentsItem> {
```

Comment:
I'm tempted to rename these types while I'm doing a breaking change anyway.

`ITableOfContents<T>` -> `TableOfContentsProps`
`ITableOfContentsLink` -> `TableOfContentsLink`

We desperately need a company-wide typescript style guide (LMK if there's any), but we usually don't prefix types with `I`, so I'd make it consistent.

What do you think?

Reply:
We don't use `I` prefix for types, rather interfaces. That `I` in `ITableOfContentsLink` type looks like been added by accident.
Some style guide might be useful, though some kind of styling already developed, at least as I've noticed. Like with your example, I would rather say that for props that first version might be used more often. Though I don't have preference here. I try to choose what I see mostly. 
Though if we do come up with some global styling and general rules, it may make things simpler.